connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConnector.java

```diff
@@ -0,0 +1,390 @@
+        Map<String, Long> partitionCounts = knownTopicPartitions.stream()
+            .collect(Collectors.groupingBy(x -> x.topic(), Collectors.counting())).entrySet().stream()
+            .collect(Collectors.toMap(x -> formatRemoteTopic(x.getKey()), x -> x.getValue()));
+        List<NewTopic> newTopics = partitionCounts.entrySet().stream()
+            .filter(x -> !knownTargetTopics.contains(x.getKey()))
+            .map(x -> new NewTopic(x.getKey(), x.getValue().intValue(), (short) replicationFactor))
+            .collect(Collectors.toList());
+        Map<String, NewPartitions> newPartitions = partitionCounts.entrySet().stream()
+            .filter(x -> knownTargetTopics.contains(x.getKey()))
+            .collect(Collectors.toMap(x -> x.getKey(), x -> NewPartitions.increaseTo(x.getValue().intValue())));
+        synchronized (this) {
+            targetAdminClient.createTopics(newTopics, new CreateTopicsOptions()).values().forEach((k, v) -> v.whenComplete((x, e) -> {
+                if (e != null) {
+                    log.warn("Could not create topic {}.", k, e);
+                } else {
+                    log.info("Created remote topic {} with {} partitions.", k, partitionCounts.get(k));
+                }
+            }));
+            targetAdminClient.createPartitions(newPartitions).values().forEach((k, v) -> v.whenComplete((x, e) -> {
+                if (e instanceof InvalidPartitionsException) {
+                    // swallow, this is normal
+                } else if (e != null) {
+                    log.warn("Could not create topic-partitions for {}.", k, e);
+                } else {
+                    log.info("Increased size of {} to {} partitions.", k, partitionCounts.get(k));
+                }
+            }));
+        }
+    }
+
+    private Set<String> listTopics(AdminClient adminClient)
+            throws InterruptedException, ExecutionException {
+        synchronized (this) {
+            return adminClient.listTopics().names().get();
+        }
+    }
+
+    private Collection<AclBinding> listTopicAclBindings()
+            throws InterruptedException, ExecutionException {
+        synchronized (this) {
+            return sourceAdminClient.describeAcls(ANY_TOPIC_ACL).values().get();
+        }
+    }
+
+    private Collection<TopicDescription> describeTopics(Collection<String> topics)
+            throws InterruptedException, ExecutionException {
+        synchronized (this) {
+            return sourceAdminClient.describeTopics(topics).all().get().values();
+        }
+    }
+
+    @SuppressWarnings("deprecation")
+    // use deprecated alterConfigs API for broker compatibility back to 0.11.0
+    private void updateTopicConfigs(Map<String, Config> topicConfigs)
+            throws InterruptedException, ExecutionException {
+        Map<ConfigResource, Config> configs = topicConfigs.entrySet().stream()
+            .filter(x -> shouldReplicateTopicConfigurationProperty(x.getKey()))
```

Comment:
`x.getKey()` is the topic name. We need to iterate over `topicConfigs.values().entries()` instead to filter out topic properties.

Reply:
great catch! I've fixed and added a unit test.